dev.sh

```diff
@@ -207,6 +207,56 @@ ___helium_validate() {
     fi
 }
 
+___helium_format() {
+    cd "$_src_dir"
+    quilt diff | "$_clang_dir/share/clang/clang-format-diff.py" -p1 -i -style=file
+}
+
+___helium_find_tidy_diff() {
+    if [ -n "$_tidy_diff_script" ]; then
+        return
+    elif command -v clang-tidy-diff >/dev/null 2>&1; then
+        _tidy_diff_script=$(command -v clang-tidy-diff)
+    elif command -v clang-tidy-diff.py >/dev/null 2>&1; then
+        _tidy_diff_script=$(command -v clang-tidy-diff.py)
+    else
+        _tidy_diff_script=$(find /opt/homebrew/Cellar/llvm -name clang-tidy-diff.py | head -1)
+    fi
+
+    if [ -z "$_tidy_diff_script" ]; then
+        echo "could not find clang-tidy-diff.py script." >&2
+        echo "ensure that you have llvm installed on your system" >&2
+        return 1
+    fi
+}
+
+___helium_strip_compile_commands() {
+    _ccmd_path="$_src_dir/out/Default/compile_commands.json"
+    [ "$_ccmd_stripped" = 1 ] && return;
+    _ccmd_stripped=1
+
+    echo "normalizing compile_commands.json, this will take a while..."
+    cp "$_ccmd_path" "$_ccmd_path.orig";
+    gsed -Ei 's/^(\s*"command": ").*?\s(\S+bin\/clang)/\1\2/g' "$_ccmd_path"
+}
+
+___helium_tidy() {
+    ___helium_find_tidy_diff || return;
+    ___helium_strip_compile_commands;
+    quilt diff | "$_tidy_diff_script" \
+        -regex '.*\.(cc|mm)' \
```

Comment:
not sure if [this](https://clang.llvm.org/extra/clang-tidy/#:~:text=file%2C%20if%20any.-,%2D%2Dheader%2Dfilter,-%3D%3Cstring%3E%20%20%20%20%20%20%20%20%20%2D%20Regular%20expression) is the right option or the right way to use it, but it'd be nice to check header files too, if they're not tidied by default, as i can't really tell what's the case here

```suggestion
        -regex '.*\.(cc|mm)' \
        -header-filter="$_src_dir" \
```